src/chess/board.cc

```diff
@@ -743,18 +743,21 @@ bool ChessBoard::IsSameMove(Move move1, Move move2) const {
       move2.to().row() != RANK_1) {
     return false;
   }
-  const bool move1_is_castling = our_pieces_.get(move1.to()) ||
-                                 abs(move1.to().col() - move1.from().col()) > 1;
-  const bool move2_is_castling = our_pieces_.get(move2.to()) ||
-                                 abs(move2.to().col() - move2.from().col()) > 1;
-  // If both moves are not castlings, then it's surely not the same move (we
-  // already checked for equality).
-  if (!move1_is_castling || !move2_is_castling) return false;
-  // It's two castlings, checking that they are castlings in the same direction.
-  if (move1.to().col() > move1.from().col()) {
-    return move2.to().col() > move2.from().col();
-  }
-  return move2.to().col() < move2.from().col();
+  // Explicitly check all legacy castling moves
+  if (move1.from() != FILE_E) return false;
+  if (move1.to().col() == FILE_A && move2.to().col() == FILE_C &&
```

Comment:
Looks good, but this one is probably better for readability:

```cpp
for (const auto& x : {{FILE_A, FILE_C}, {FILE_C, FILE_A}, {FILE_H, FILE_G}, {FILE_G, FILE_H}}) {
  if (....) {
    return true;
  }
}
```

Also if `if` is multiline, curly braces are necessary.

And also, I think it's safer to check for own pieces than for rooks, as rooks() also contain opponent's rooks and capture of a opponent rook is a legal move. (not by jumping from file E to A, but still "more legal").

Reply:
Actually, I'm not sure we even need to check move legality here or just the representation. So maybe remove both checks for king and rook?